server-node-new/app.js

```diff
@@ -0,0 +1,197 @@
+const createError = require("http-errors");
+const express = require("express");
+const { join } = require("path");
+const cookieParser = require("cookie-parser");
+const logger = require("morgan");
+
+const indexRouter = require("./routes/index");
+const pingRouter = require("./routes/ping");
+
+const { json, urlencoded } = express;
+
+var app = express();
+
+// Database Setup
+
+const mongoose = require("mongoose");
+const mongoDB = "mongodb://localhost/test";
+mongoose.connect(mongoDB, { useNewUrlParser: true, useUnifiedTopology: true });
+const db = mongoose.connection;
+db.once("open", function(){
+  console.log("Database connected successfully");
+}).on("error", function(err){
+  console.log("Error", err);
+});
+
+// Models
+
+const userSchema = new mongoose.Schema({
+  userName:         { type: String, index: true, unique: true, sparse: true }, // not sure if this is the standard implementation to allow null to not be unique
```

Comment:
this index might be giving you the index deprecation issue

Reply:
https://stackoverflow.com/a/40760224/14204824

probably need to create a partial index after creating the schema to ensure a index and unique value

Reply:
https://stackoverflow.com/a/44641358/14204824